pkg/policy/api/rule.go

```diff
@@ -101,39 +102,16 @@ func (r *Rule) MarshalJSON() ([]byte, error) {
 		buf.Write(jsonValue)
 	}
 
-	if r.Ingress != nil {
-		jsonValue, err := json.Marshal(r.Ingress)
+	v := reflect.Indirect(reflect.ValueOf(r))
+	types := v.Type()
+	for i := 2; i < v.NumField(); i++ {
+		jsonValue, err := json.Marshal(v.Field(i).Interface())
 		if err != nil {
 			return nil, err
 		}
-		buf.Write([]byte(`,"ingress":`))
-		buf.Write(jsonValue)
-	}
-
-	if r.Egress != nil {
-		jsonValue, err := json.Marshal(r.Egress)
-		if err != nil {
-			return nil, err
-		}
-		buf.Write([]byte(`,"egress":`))
-		buf.Write(jsonValue)
-	}
-
-	if r.Labels != nil {
-		jsonValue, err := json.Marshal(r.Labels)
-		if err != nil {
-			return nil, err
-		}
-		buf.Write([]byte(`,"labels":`))
-		buf.Write(jsonValue)
-	}
-
-	if r.Description != "" {
-		jsonValue, err := json.Marshal(r.Description)
-		if err != nil {
-			return nil, err
-		}
-		buf.Write([]byte(`,"description":`))
+		buf.Write([]byte(`,"`))
+		buf.WriteString(types.Field(i).Name)
```

Comment:
I think this is not ideal because it's ignoring if the json tags. For example, the field name is different than the json tag, the "omitempty" is not being considered.

Reply:
My bad. That's what I planned to do, but I guess I just forgot about it at some point :disappointed:

Fixed.